test/integration/util/util.go

```diff
@@ -225,6 +225,25 @@ func AddTaintToNode(cs clientset.Interface, nodeName string, taint v1.Taint) err
 	return err
 }
 
+// RemoveTaintOffNode removes taints off a specific node
+func RemoveTaintOffNode(cs clientset.Interface, nodeName string, taint v1.Taint) error {
+	node, err := cs.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	copy := node.DeepCopy()
+	taints := []v1.Taint{}
+	for _, t := range copy.Spec.Taints {
+		if t == taint {
```

Comment:
This looks problematic as the passed in `taint` may not have `TimeAdded` field.

---

https://github.com/kubernetes/kubernetes/blob/bded41a817a84cd7e51fa535ad6def319a892a3c/staging/src/k8s.io/api/core/v1/taint.go#L23-L25 

can be used if you don't care about the taint's value.

Reply:
makes a lot of sense, thanks for spotting!